src/dune_engine/process.ml

```diff
@@ -360,23 +360,64 @@ let pp_id id =
   let open Pp.O in
   Pp.char '[' ++ Pp.tag User_message.Style.Id (Pp.textf "%d" id) ++ Pp.char ']'
 
-module Exit_status = struct
+module Exit_status : sig
   type error =
     | Failed of int
     | Signaled of string
 
   type t = (int, error) result
 
+  val handle_verbose :
+       ('a, error) result
+    -> id:int
+    -> purpose:purpose
+    -> output:string
+    -> command_line:User_message.Style.t Pp.t
+    -> dir:With_directory_annot.payload option
+    -> 'a
+
+  val handle_non_verbose :
+       ('a, error) result
+    -> verbosity:Scheduler.Config.Display.verbosity
+    -> purpose:purpose
+    -> output:string
+    -> prog:string
+    -> command_line:string
+    -> dir:With_directory_annot.payload option
+    -> has_unexpected_stdout:bool
+    -> has_unexpected_stderr:bool
+    -> 'a
+end = struct
+  type error =
+    | Failed of int
+    | Signaled of string
+
+  type t = (int, error) result
+
+  type output =
+    { with_color : User_message.Style.t Pp.t
+    ; without_color : string
+    }
+
   let parse_output = function
     | "" -> None
     | s ->
       Some
-        (Pp.map_tags (Ansi_color.parse s) ~f:(fun styles ->
-             User_message.Style.Ansi_styles styles))
+        (let with_color =
+           Pp.map_tags (Ansi_color.parse s) ~f:(fun styles ->
+               User_message.Style.Ansi_styles styles)
+         in
+         let without_color = Ansi_color.strip s in
```

Comment:
Just to let you know, I changed the code to `Ansi_color.strip` here. Feels lighter than going through the formatter machinery.